### PR TITLE
Address repository audit hardening

### PR DESCRIPTION
**Pull request by ar27111994**
## Summary

Addresses the post-merge repository audit with focused hardening, safer defaults, and updated documentation.

### Security and runtime hardening
- reject mirror multi-file artifact path traversal outside the raw mirror root
- add guarded HTTP helpers with origin allowlists, timeouts, and response byte limits
- use guarded fetches for official index content, official page summaries, raw GitHub file content, npm metadata, and PyPI metadata
- validate and normalize PyPI metadata field-by-field before repository URL extraction
- reset GitHub process-local rate-limit and health-update state during CLI bootstrap
- resolve the VS Code settings path lazily after `.env` loading and runtime config reset

### Lifecycle and CLI behavior
- make `wire <host>` default to preview; require explicit `--apply` or `--reset` for mutations
- run `recommend report` explicitly in workspace orchestration instead of relying on `discover select` side effects
- fix install batching so missing progress state is not treated as completion
- centralize CLI option parsing and reject flag-looking values as missing option values
- document `doctor` and default `recommend report` behavior

### Source utilization and docs
- attempt guarded summaries for docs, registry, and marketplace reference sources
- distinguish active, reference-only, and dormant source utilization
- update roadmap and implementation-plan status so remaining roadmap-scale work is accurately tracked
- align package description with current host support

### Tests
- add regression tests for dotenv duplicate/multiline behavior
- add CLI option missing-value tests
- add safe mirror path and PyPI normalization tests
- add VS Code settings read/patch tests

## Validation

- `npm run typecheck`
- `npm run lint`
- `npm run format:check`
- `npm run build`
- `npm test`
- `npm run smoke:cli`
- `npm run benchmark:scan`
- `npm run quality:detection`
- `npm run quality:policy`
- `npm run validate:recommendations`
- `git diff --check`
- `node ./dist/cli.js setup doctor --host --other` (fails with the expected missing-value message)
- `node ./dist/cli.js wire cursor` (defaults to preview)

## Release Notes

* **New Features**
  * Wire command defaults to preview mode; use `--apply` or `--reset` for mutations
  * Workspace runs now explicitly invoke recommendations after discovery

* **Bug Fixes**
  * Safe mirror artifact paths against traversal attacks
  * Guarded HTTP fetching with origin allowlists and byte limits
  * PyPI metadata field validation and normalization
  * Process-local GitHub state reset between CLI invocations
  * Install batching progress state handling
  * Lazy VS Code settings path resolution for .env overrides
  * CLI option parsing rejects invalid flag tokens

* **Documentation & Tests**
  * Updated lifecycle and command documentation
  * Added tests for dotenv parsing, CLI options, and security hardening

### CODE REVIEW

File: src/official-index.ts [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 import type { AssetStatus, AuthorityTier } from "./types.js";
 
 const OFFICIAL_INDEX_USER_AGENT = "agent-harness";
 const MAX_LIST_ITEMS = 5;
+const MAX_OFFICIAL_INDEX_PAGE_BYTES = 1_000_000;
+const OFFICIAL_INDEX_ALLOWED_ORIGINS = [
+  "https://officialskills.sh",
+  "https://raw.githubusercontent.com",
+  "https://docs.github.com",
+  "https://docs.anthropic.com",
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/official-index.ts:11]: Addressed in `5479712`. Expanded the reference-source summary allowlist to include the configured package registry reference origins `https://www.npmjs.com` and `https://pypi.org`, in addition to the existing docs/marketplace origins.

File: src/lib/http.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+export async function fetchTextWithGuards(
+  url: string,
+  options: FetchWithGuardsOptions = {},
+): Promise<string | null> {
+  try {
+    const parsedUrl = assertAllowedHttpUrl(url, options.allowedOrigins ?? []);
+    const response = await fetchWithTimeout(
+      parsedUrl.toString(),
+      { headers: options.headers },
+      options.timeoutMs ?? DEFAULT_TIMEOUT_MS,
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/lib/http.ts:56]: Addressed in `5479712`. Guarded text fetches now call fetch with `redirect: "error"`, so automatic redirects cannot follow an allowed-origin URL to an unallowed origin. Added regression coverage asserting the guarded helper disables automatic redirects.

File: src/wire.ts [hunk 3]
```diff
@@ -75,7 +75,7 @@ function getWireMode(args: string[]): "preview" | "apply" | "reset" {
     return "preview";
   }
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/wire.ts:77]: Addressed in `5479712`. Added an explicit `--apply` branch in `getWireMode` and regression coverage confirming default preview, explicit preview/apply/reset, and conflicting mode handling.

File: src/lib/http.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  maxBytes?: number;
+  timeoutMs?: number;
+}
+
+const DEFAULT_TIMEOUT_MS = 10_000;
+const DEFAULT_MAX_RESPONSE_BYTES = 1_000_000;
+
+/**
+ * Performs an HTTP(S) fetch with a timeout after validating the destination
+ * origin against an explicit allowlist.
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/lib/http.ts]: Addressed in `5479712`. Updated the `fetchWithTimeout` doc comment to describe only its actual timeout behavior and explicitly note that allowlist enforcement belongs to the higher-level guarded fetch helpers.

File: src/lib/http.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  timeoutMs = DEFAULT_TIMEOUT_MS,
+): Promise<Response> {
+  const controller = new AbortController();
+  const timer = setTimeout(() => controller.abort(), timeoutMs);
+
+  try {
+    return await fetch(url, { ...options, signal: controller.signal });
+  } finally {
+    clearTimeout(timer);
+  }
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/lib/http.ts:40]: Addressed in `828c1b6`. `fetchWithTimeout` now preserves caller-provided abort signals by propagating upstream aborts into the timeout controller and removing the listener in `finally`. Added regression coverage for caller-driven abort propagation.

File: src/package-registries.ts [hunk 6]
```diff
@@ -84,21 +58,104 @@ export async function fetchPypiPackageMetadata(
...
+      version:
+        typeof data.info.version === "string" ? data.info.version : undefined,
+      keywords:
+        typeof data.info.keywords === "string" ? data.info.keywords : undefined,
+      package_url:
+        typeof data.info.package_url === "string"
+          ? data.info.package_url
+          : undefined,
+    },
+  };
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/package-registries.ts:133]: Addressed in `828c1b6`. PyPI URL fields are now normalized through URL parsing for `home_page`, `package_url`, and `project_urls`; repository extraction now requires an actual `github.com` hostname with owner/repo path instead of substring matching. Added regression coverage for the query-string trap case.

File: src/discover.ts [hunk 7]
```diff
@@ -451,19 +459,34 @@ async function writeSourceUtilizationReport(
...
+      operational: operationalEntries.length > 0,
+      harvestedEntries: sourceEntries.length,
+      operationalEntries: operationalEntries.length,
+      status:
+        operationalEntries.length > 0
+          ? "active"
+          : sourceEntries.length > 0
+            ? "reference-only"
+            : "dormant",
     };
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/discover.ts:490]: Addressed in `828c1b6`. `dormantSourceCount` now counts only sources whose per-source status is exactly `dormant`, so `reference-only` sources no longer inflate the aggregate dormant count.

File: src/lib/http.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/lib/http.ts:40]: Verified current code. The `discover.ts`, `http.ts`, `package-registries.ts`, and `wire.ts` findings were already addressed in earlier commits. The remaining `cli.ts` nitpick was valid, so `5b76ba3` restores `console.error(error)` in the top-level catch handler to preserve Error stacks while keeping `process.exitCode = 1`. Validation passed: `npm run typecheck`, `npm run lint`, `npm run format:check`, `npm run build`, and `npm test`.

File: src/pipeline.ts [hunk 8]
```diff
@@ -130,7 +132,7 @@ async function installBundleBatches(
         join(projectRoot, ...INSTALL_PROGRESS_STATE_PATH),
       );
       const bundleState = progressState?.bundles[bundleId];
-      if (!bundleState || bundleState.remainingAssets <= 0) {
+      if (bundleState && bundleState.remainingAssets <= 0) {
         break;
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/pipeline.ts:148]: Addressed in `1a5632e`. The batching loop now fails fast when `install bundle` does not produce `state/install/progress.json` or does not include progress for the requested bundle, instead of retrying up to the max batch count with a generic timeout-style error.

File: src/mirror.ts [hunk 9]
```diff
@@ -772,17 +783,25 @@ function sanitizeMirrorId(value: string): string {
...
+  const resolvedTarget = resolve(resolvedRoot, relativePath);
+  const relativeTarget = relative(resolvedRoot, resolvedTarget);
 
-  if (optionIndex === -1) {
-    return undefined;
+  if (
+    relativeTarget.length === 0 ||
+    relativeTarget.startsWith("..") ||
+    isAbsolute(relativeTarget)
+  ) {
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/mirror.ts:799]: Addressed in `1a5632e`. Tightened the traversal check to reject only the exact parent segment (`..`) or paths starting with `..${path.sep}` after normalization, while still rejecting absolute resolved escapes. Added a regression assertion that an in-root `..foo/SKILL.md` path is accepted.
